Dockerfile.slim

```diff
@@ -1,33 +1,47 @@
-FROM alpine:3.12
+FROM python:3.7-alpine3.12
 LABEL maintainer="sig-platform@spinnaker.io"
 
-# KUBECTL_VERSION kept one minor version behind latest to maximise compatibility overlap
-ENV KUBECTL_VERSION v1.18.10
-
-RUN apk --no-cache add --update bash wget unzip openjdk11 'python2>2.7.9' && \
-  wget -nv https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.zip && \
-  unzip -qq google-cloud-sdk.zip -d /opt && \
-  rm google-cloud-sdk.zip && \
-  CLOUDSDK_PYTHON="python2.7" /opt/google-cloud-sdk/install.sh --usage-reporting=false --bash-completion=false --additional-components app-engine-java app-engine-go && \
-  rm -rf ~/.config/gcloud
-
-RUN wget https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl && \
-  chmod +x kubectl && \
-  mv ./kubectl /usr/local/bin/kubectl
-
-RUN wget https://amazon-eks.s3.us-west-2.amazonaws.com/1.16.8/2020-04-16/bin/linux/amd64/aws-iam-authenticator && \
-  chmod +x ./aws-iam-authenticator && \
-  mv ./aws-iam-authenticator /usr/local/bin/aws-iam-authenticator && \
-  ln -sf /usr/local/bin/aws-iam-authenticator /usr/local/bin/heptio-authenticator-aws
-
-RUN apk -v --update add py-pip && \
-  pip install --upgrade awscli==1.18.152 s3cmd==2.0.2 python-magic && \
-  apk -v --purge del py-pip && \
-  rm /var/cache/apk/*
-
-ENV PATH "$PATH:/usr/local/bin/"
-
-ENV PATH=$PATH:/opt/google-cloud-sdk/bin/
+# KUBECTL_RELEASE kept one minor version behind latest to maximise compatibility overlap
+ENV KUBECTL_RELEASE=1.18.10
+ENV AWS_CLI_VERSION=1.18.152
+ENV AWS_CLI_S3_CMD=2.0.2
+ENV AWS_AIM_AUTHENTICATOR_VERSION=0.4.0
+ENV GOOGLE_CLOUD_SDK_VERSION=313.0.1
+ENV ECR_TOKEN_VERSION=v1.0.2
```

Comment:
Nice move, surfacing these higher up!

Reply:
Arrggg. AWS_AIM_AUTHENTICATOR_VERSION variable is the older version which doesn't support IRSA and was* fixed previously in #4696

Reply:
This is intentional.  We encountered some rather nasty bugs in 0.5.0 that cause slow-downs in large environments.  0.5.2 SLIGHTLY fixed this but was substantial enough of an impact that we recommended rolling back for all our customers.  The recommendation was been "DON'T" use aws-iam-authenticator but instead the aws cli for those cases.  See: https://github.com/kubernetes-sigs/aws-iam-authenticator/issues/347 for reference.  The AWS CLI should support this natively instead of using aws-iam-authenticator